resources.js

```diff
@@ -1,15 +1,86 @@
 export class Media {
-  // TODO
+  constructor(title){
+    this._title = title;
+    this._isCheckedOut = false;
+    this._ratings = [];
+  }
+  
+  get title(){
+    return this._title;
+  }
+  
+  get isCheckedOut(){
+    return this._isCheckedOut;
+  }
+  
+  get ratings(){
+    return this._ratings;
+  }
+
+  getAverageRating(){
+    const curRatings = this._ratings;
+    const curRatingsLength = curRatings.length;
+    const curRatingsSum = curRatings.reduce((acc, curValue) => acc + curValue);
+    const avrRating = curRatingsSum / curRatingsLength;
+    
+    return avrRating;
+  }
+
+  toggleCheckOutStatus(){
+    this._isCheckedOut = !this._isCheckedOut;
+    return this._isCheckedOut;
```

Comment:
This wasn't required in the README but I don't mind.

Reply:
I thought the function should toggle the CheckOut status opposite.

Reply:
Yes, it should. I mean returning it's value is redundant.